scipy/signal/_spectral_py.py

```diff
@@ -57,7 +75,18 @@ def lombscargle(x,
     Raises
     ------
     ValueError
-        If the input arrays `x` and `y` do not have the same shape.
+        If any of the input arrays x, y, freqs, or weights are not 1D. Or, if the input 
+        arrays x, y, and weights do not have the same shape.
```

Comment:
All of the numpy inputs must be real. I'll check `isrealobj` for all.

Reply:
I changed this to `iscomplexobj` because `isrealobj` just calls `not iscomplexobj`, so you end up with two `not`s in the condition canceling each other.